store rank 0 instance URIs in enclosing instance KVS

garlick opened the pull request:
This PR is an attempt to address #1422 by writing an instance's rank 0 FLUX_URI and a derived ssh:// URI to `lwj.X.X.X.flux.local-uri` and `lwj.X.X.X.flux.remote-uri` in the enclosing instance.

I need to provide a test but thought I'd post early to get any feedback from @trws.

trws commented:
That sounds perfect!

On 5 Apr 2018, at 7:31, Jim Garlick wrote:

> This PR is an attempt to address #1422 by writing an instance's rank 0 
> FLUX_URI and a derived ssh:// URI to `lwj.X.X.X.flux.local-uri` and 
> `lwj.X.X.X.flux.remote-uri` in the enclosing instance.
>
> I need to provide a test but thought I'd post early to get any 
> feedback from @trws.
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/flux-framework/flux-core/pull/1429
>
> -- Commit Summary --
>
>   * wreck: set FLUX_JOB_KVSPATH
>   * rc1: inform enclosing instance of useful URIs
>
> -- File Changes --
>
>     M configure.ac (6)
>     M etc/Makefile.am (3)
>     A etc/rc1.d/01-enclosing-instance (16)
>     M src/modules/wreck/wrexecd.c (1)
>
> -- Patch Links --
>
> https://github.com/flux-framework/flux-core/pull/1429.patch
> https://github.com/flux-framework/flux-core/pull/1429.diff
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/flux-framework/flux-core/pull/1429

garlick commented:
Great.  I'll add a `flux wreck uri` command to make it easier to access.

grondo commented:
Nice!  Is there a requirement here to add to our requirements/use case doc?

trws commented:
It might be good for a use-case doc.  This is a good step one for 
working with nested flux instances, especially when users need them for 
job step execution in regular jobs, we’re doing a lot of that for the 
splash app and I found myself wishing for this frequently.

On 5 Apr 2018, at 9:49, Mark Grondona wrote:

> Nice!  Is there a requirement here to add to our requirements/use case 
> doc?
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/flux-framework/flux-core/pull/1429#issuecomment-379002142

garlick commented:
OK, subcommand looks like this:
```
$ flux wreck uri
    ID NTASKS STATE     FLUX_URI                                 COMMAND
     1      1 exited    -                                        hostname
     2      1 exited    -                                        hostname
     3      1 exited    ssh://jimbo//tmp/flux-qCbPz5             flux
```

trws commented:
That looks good.  Would it be possible to get a variant where I provide 
a job-id and it prints just the uri for that job? It would help me with 
scripting helpers if that isn’t too much trouble.

On 5 Apr 2018, at 10:16, Jim Garlick wrote:

> OK, subcommand looks like this:
> ```
> $ flux wreck uri
>     ID NTASKS STATE     FLUX_URI                                 
> COMMAND
>      1      1 exited    -                                        
> hostname
>      2      1 exited    -                                        
> hostname
>      3      1 exited    ssh://jimbo//tmp/flux-qCbPz5             flux
> ```
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/flux-framework/flux-core/pull/1429#issuecomment-379010340

garlick commented:
Yes, good idea.

garlick commented:
OK, I added a `--bare` option to print just the URI for a single job.

Added a couple tests also.  This is probably close to ready if travis agrees.

> Nice! Is there a requirement here to add to our requirements/use case doc?

I've made a note to return to this after the splash fire drill - probably!  And it may be that we'll want something that works for deeper recursion long term...

garlick commented:
Dropped the lua `f:kvs_rawget()` function, which had no tests of its own, because it's not really necessary.  I just put the values in the kvs with `flux kvs put --json` instead and the the old `f:kvs_get()` works and this PR becomes smaller.

garlick commented:
Rebased.

garlick commented:
Thanks!

trws commented:
This looks really great, looking forward to testing it, thanks!
